apps/web/src/components/AccountSelectorModal/DeriveMnemonicAccountModal/DeriveMnemonicAccountModal.test.tsx

```diff
@@ -0,0 +1,67 @@
+import { mockMnemonicAccount } from "@umami/core";
+import { type UmamiStore, addTestAccount, makeStore, useDeriveMnemonicAccount } from "@umami/state";
+
+import { DeriveMnemonicAccountModal } from "./DeriveMnemonicAccountModal";
+import { act, renderInModal, screen, userEvent, waitFor } from "../../../testUtils";
+
+let store: UmamiStore;
+
+jest.mock("@umami/state", () => ({
+  ...jest.requireActual("@umami/state"),
+  useDeriveMnemonicAccount: jest.fn(),
+}));
+
+const mockDeriveMnemonicAccount = jest.fn();
+
+beforeEach(() => {
+  store = makeStore();
+  addTestAccount(store, mockMnemonicAccount(0));
+
+  jest.mocked(useDeriveMnemonicAccount).mockImplementation(() => mockDeriveMnemonicAccount);
+});
+
+describe("<DeriveMnemonicAccountModal />", () => {
+  it("renders the NameAccountModal with correct subtitle", async () => {
+    const account = mockMnemonicAccount(0);
+    await renderInModal(<DeriveMnemonicAccountModal account={account} />, store);
+
+    await waitFor(() => {
+      expect(screen.getByText("Name Your Account")).toBeVisible();
+    });
+
+    expect(
+      screen.getByText(`Name the new account derived from seedphrase ${account.seedFingerPrint}`)
+    ).toBeVisible();
+  });
+
+  it("handles name submission and opens confirm password modal", async () => {
+    const user = userEvent.setup();
+
+    const account = mockMnemonicAccount(0);
+    await renderInModal(<DeriveMnemonicAccountModal account={account} />, store);
+    await act(() => user.type(screen.getByLabelText("Account name"), "Test Account"));
+    await act(() => user.click(screen.getByRole("button", { name: "Continue" })));
+
+    await waitFor(() => {
+      expect(screen.getByTestId("master-password-modal")).toBeVisible();
+    });
+  });
+
```

Comment:
Maybe we can also check the case with incorrect password

And that after successful deriving, the modal will be closed